testnet-contracts/contracts/Bank.sol

```diff
@@ -0,0 +1,96 @@
+pragma solidity ^0.5.0;
+
+import "../../node_modules/openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./BankToken.sol";
+
+/**
+ * @title Bank
+ * @dev Manages the deployment and minting of ERC20 compatible BankTokens
+ **/
+
+contract Bank {
+
+    using SafeMath for uint256;
+
+    mapping(address => bool) public bankTokens;
+    uint256 public numbTokens;
+
+    event LogBankTokenDeploy(address _token);
+
+    event LogBankTokenMint(
+        address _token,
+        string _symbol,
+        uint256 _amount,
+        address _beneficiary
+    );
+
+    constructor () public {
+        numbTokens = 0;
+    }
+
+    /*
+     * @dev: Delivers bank tokens
+     *
+     * @param _token: bank token contract address, address(0) indicates new token
+     * @param _symbol: bank token symbol
+     * @param _amount: number of bank tokens to be delivered
+     * @param _beneficiary: recipient of the minted tokens
+     */
+     function deliver(
+        address _token,
+        string memory _symbol,
+        uint256 _amount,
+        address _beneficiary
+    )
+        internal
+    {
+        // If no token address, deploy a new bank token
```

Comment:
Hmm I think it may be better to make deployment happen outside of the deliver method rather than inside, ie, deliver always fails if the token has not been deployed yet?